### PR TITLE
Mediborg's Amputation Adventure - Emag Edition

### PR DESCRIPTION
**Pull request by Ivniinvi**
## About The Pull Request
You can now emag Mediborg's Amputation Adventure, instead of taking one limb it will take all of them, and dispense 20-30 tokens if you choose to do so.

## Why It's Good For The Game
The other two arcade machines have emag functions, why not this one?

## Changelog
:cl:
add: Mediborg's Amputation Adventure can now be emagged.
/:cl:

### CODE REVIEW

**Comment by tiramisuapimancer** [comment]
Can you slam a [Cabinet Man](https://www.youtube.com/watch?v=iZ_P3F3nA1U) reference into this? I'll be your best friend!

File: code/game/machinery/computer/arcade.dm [hunk 1]
```diff
@@ -1245,23 +1245,51 @@ GLOBAL_LIST_INIT(arcade_prize_pool, list(
...
+			to_chat(c_user, "<span class='userdanger'>Robotic arms shoot out of the machine, remove all your limbs, and suck them in!</span>")
+			playsound(loc, 'sound/weapons/slice.ogg', 25, 1, -1)
+			for(var/X in c_user.bodyparts)
+				var/obj/item/bodypart/BP = X
+				if(BP.body_part != HEAD && BP.body_part != CHEST)
+					if(BP.dismemberable)
+						BP.dismember()
+						qdel(BP)
+			playsound(loc, 'sound/arcade/win.ogg', 50, 1, extrarange = -3, falloff_exponent = 10)
+			for(var/i=1; i<=rand(20,30); i++)
```
Comment (Archanial) [inline code/game/machinery/computer/arcade.dm]:
```suggestion
			for(var/i=1 to rand(20, 30))
```

File: code/game/machinery/computer/arcade.dm [hunk 2]
```diff
@@ -1245,23 +1245,51 @@ GLOBAL_LIST_INIT(arcade_prize_pool, list(
...
+				prizevend(user)
+		else
+			to_chat(c_user, "<span class='notice'>You (wisely) decide against putting your hand in the machine.</span>")
```
Comment (Archanial) [inline code/game/machinery/computer/arcade.dm:1265]: put that message after everything and return after completing action, instead.

File: code/game/machinery/computer/arcade.dm [hunk 3]
```diff
@@ -1245,23 +1245,51 @@ GLOBAL_LIST_INIT(arcade_prize_pool, list(
...
+			to_chat(c_user, "<span class='userdanger'>The guillotine drops on your arm, and the machine sucks it in!</span>")
+			playsound(loc, 'sound/weapons/slice.ogg', 25, 1, -1)
+			var/which_hand = BODY_ZONE_L_ARM
+			if(!(c_user.active_hand_index % 2))
+				which_hand = BODY_ZONE_R_ARM
+			var/obj/item/bodypart/chopchop = c_user.get_bodypart(which_hand)
+			chopchop.dismember()
+			qdel(chopchop)
+			playsound(loc, 'sound/arcade/win.ogg', 50, 1, extrarange = -3, falloff_exponent = 10)
+			for(var/i=1; i<=rand(3,5); i++)
```
Comment (Archanial) [inline code/game/machinery/computer/arcade.dm]:
```suggestion
			for(var/i=1 to rand(3, 5))
```

**Review by Archanial — CHANGES_REQUESTED** [review]

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**